src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -165,14 +165,11 @@ private IEnumerable<RazorDiagnostic> Validate()
 
             if (string.IsNullOrWhiteSpace(_name))
             {
-                if (_indexerNamePrefix == null)
```

Comment:
Since we validate as part of Build(), the diagnostics are same for both the expected and actual descriptors which makes the tests pass. I'll file an issue to refactor our tests to test the diagnostics separately.